src/ml/neural_net/mps_layer_instance_norm_data_loader.mm

```diff
@@ -0,0 +1,202 @@
+                                            length:sizeof(float) * _numberOfFeatureChannels
+                                           options:MTLResourceStorageModeManaged];
+
+    _gammaVelocityBuffer = [dev newBufferWithBytes:zeros_ptr
+                                            length:sizeof(float) * _numberOfFeatureChannels
+                                           options:MTLResourceStorageModeManaged];
+
+    _betaMomentumBuffer = [dev newBufferWithBytes:zeros_ptr
+                                           length:sizeof(float) * _numberOfFeatureChannels
+                                          options:MTLResourceStorageModeManaged];
+
+    _betaVelocityBuffer = [dev newBufferWithBytes:zeros_ptr
+                                           length:sizeof(float) * _numberOfFeatureChannels
+                                          options:MTLResourceStorageModeManaged];
+
+    _movingMeanBuffer = [dev newBufferWithBytes:zeros_ptr
+                                         length:sizeof(float) * _numberOfFeatureChannels
+                                        options:MTLResourceStorageModeManaged];
+
+    _movingVarianceBuffer = [dev newBufferWithBytes:ones_ptr
+                                             length:sizeof(float) * _numberOfFeatureChannels
+                                            options:MTLResourceStorageModeManaged];
+    
+    _adamGamma = [[MPSNNOptimizerAdam alloc] initWithDevice:dev
+                                              learningRate:0.001f];
+        
+    _adamBeta = [[MPSNNOptimizerAdam alloc] initWithDevice:dev
+                                             learningRate:0.001f];
+      
+    _gammaVector = [[MPSVector alloc] initWithBuffer:_gammaBuffer
+                                          descriptor:_vDesc];
+
+    _gammaMomentumVector = [[MPSVector alloc] initWithBuffer:_gammaMomentumBuffer
+                                                  descriptor:_vDesc];
+
+    _gammaVelocityVector = [[MPSVector alloc] initWithBuffer:_gammaVelocityBuffer
+                                                  descriptor:_vDesc];
+
+    _betaVector = [[MPSVector alloc] initWithBuffer:_betaBuffer
+                                         descriptor:_vDesc];
+
+    _betaMomentumVector = [[MPSVector alloc] initWithBuffer:_betaMomentumBuffer
+                                                 descriptor:_vDesc];
+
+    _betaVelocityVector = [[MPSVector alloc] initWithBuffer:_betaVelocityBuffer
+                                                 descriptor:_vDesc];
+
+
+    _state = [[MPSCNNNormalizationGammaAndBetaState alloc] initWithGamma:_gammaBuffer
+                                                                    beta:_betaBuffer];
+  }
+  return self;
+}
+
+- (void) updateNumberOfStyles:(NSUInteger)styles {
+  _currentStyle = 0;
+  _styles = styles;
+}
+
+- (void) updateCurrentStyle:(NSUInteger)style {
+  _currentStyle = style;
+}
+
+- (NSUInteger) getCurrentStyle {
```

Comment:
In idiomatic obj-c code, we should just use the `currentStyle` property. ObjC auto-generates this getter and setter for you.

Reply:
done